src/coreclr/jit/utils.cpp

```diff
@@ -2814,6 +2815,11 @@ uint32_t BitOperations::PopCount(uint32_t value)
     value = (((value + (value >> 4)) & c3) * c4) >> 24;
 
     return value;
+#elif HOST_ARM64
+    return _CountOneBits(value);
+#else
+    return __popcnt(value);
```

Comment:
This isn't safe. It will always emit `popcnt` which requires SSE4.2

We'd need a cached CPUID check and a branch to use it, falling back to the bit twiddling logic if unsupported.